bottery/bottery.py

```diff
@@ -104,5 +104,5 @@ def run(self, server_port):
         self.loop.run_forever()
 
     def stop(self):
-        self.session.close()
-        self.loop.close()
+        asyncio.gather(self.session.close())
```

Comment:
Why not `self.loop.run_until_complete` here?

Reply:
Because `run_until_complete` will try to invoke `.run_forever` on existing (and in some cases already running) loop. It causes an exception `RuntimeError: This event loop is already running`.

Reply:
Some cases here – I mean the situation with exception handler from the #179.

Reply:
> Because `run_until_complete` will try to invoke `.run_forever` on existing (and in some cases already running) loop. It causes an exception `RuntimeError: This event loop is already running`.

This sounds strange, `self.session.close()` is just an async method, execute it in a running loop shouldn't be a problem.

Reply:
`run_until_complete` won't just add a task to the currently running loop. Instead, it will try to add a task and **run a loop until it is executed**. 

![image](https://user-images.githubusercontent.com/3984307/47446676-086d3b80-d7bc-11e8-87bf-ecbe8c50ad77.png)

This `run_forever` call will raise an exception because loop is already running.

![image](https://user-images.githubusercontent.com/3984307/47446722-1c18a200-d7bc-11e8-9a9b-7482d2c42666.png)

What we can do is to just create a task and attach it to the currently running event loop, using `self.loop.create_task(self.session.close())`.